pkg/virtctl/suspend-resume/suspend.go

```diff
@@ -0,0 +1,174 @@
+/*
+ * This file is part of the KubeVirt project
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ *
+ * Copyright 2019 Red Hat, Inc.
+ *
+ */
+
+package suspend
+
+import (
+	"fmt"
+	"strings"
+
+	"github.com/spf13/cobra"
+
+	v1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/client-go/tools/clientcmd"
+
+	"kubevirt.io/client-go/kubecli"
+	"kubevirt.io/kubevirt/pkg/virtctl/templates"
+)
+
+const (
+	COMMAND_SUSPEND   = "suspend"
+	COMMAND_RESUME    = "resume"
+	COMMAND_VM_SHORT  = "vm"
+	COMMAND_VM_LONG   = "virtualmachine"
+	COMMAND_VMI_SHORT = "vmi"
+	COMMAND_VMI_LONG  = "virtualmachineinstance"
+)
+
+func NewSuspendCommand(clientConfig clientcmd.ClientConfig) *cobra.Command {
+	cmd := &cobra.Command{
+		Use:     "suspend vm|vmi (VM)|(VMI)",
```

Comment:
I thought we only would run "suspend on VM" and not VMI, as mostly virtctl commands are used mostly with VM and not VMI.

Feels like an exception for SUSPEND/RESUME for VMI, when using VIRTCTL.

I say this as VIRTCTL help mostly talks about "Virtual Machine/VM" and not much of "VMI".

@slintes: I know we discussed this already, but just wanted to add what I thought here and not because I disagree. :)

Reply:
Some commands (start/stop/restart) don't make sense on VMI, other do (console/vnc).

Suspend/resume makes more sense on the VMI than VM IMHO. That's why I decided to implement it on the VMI API only.
Just for more comfort, virtctl will work on VMs too. But internally virtctl is looking for the VMI's name and calling the VMI's API.
Nice (IMHO) side effect: the syntax is more aligned with kubectl, since almost all kubectl commands are like `kubectl <verb> <resourceType>[/ ]<resourceName>`